pkg/virt-api/api.go

```diff
@@ -779,9 +780,31 @@ func (app *virtAPIApp) createValidatingWebhook() error {
 }
 
 func (app *virtAPIApp) mutatingWebhooks() []admissionregistrationv1beta1.Webhook {
+	vmPath := vmMutatePath
 	vmiPath := vmiMutatePath
 	migrationPath := migrationMutatePath
 	webHooks := []admissionregistrationv1beta1.Webhook{
+		{
+			Name: "virtualmachines-mutator.kubevirt.io",
+			Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+				Operations: []admissionregistrationv1beta1.OperationType{
+					admissionregistrationv1beta1.Create,
```

Comment:
I am not familiar with this structure really, but is there Edit action too?

Reply:
There is update. Do we want to set the default also on update? The only case on which we should (and can) take action is the admin reset the machine type to "". But in that case, if we override with the default machine type, the admin is no longer capable to reset the machine type to "" (there could be legitimate reasons). Hence I believe we should do only on create.

Reply:
There are no legitimate reasons I know of to set the machine type to "". You always need to pass some machine type to qemu when starting the VM. The whole defaulting is trying to make sure it is deterministic across upgrades and restarts.

Reply:
OK, works for me. Setting the field to "" could be interpreted as "reset to kubevirt defaults". Changing.